spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,528 @@
+----------------------------- MODULE fastsyncL2 -----------------------------
+(*
+ L2 specification of the fast sync protocol.
+ 
+ In this specification, we have two parties:
+    - the node (state machine) that is doing fastsync
+    - the peersState (the peers with which node interacts to download blocks)
+
+We assume a system in which one node is trying to sync with the blockchain 
+(replicated state machine) by downloading blocks and executing transactions 
+(part of the block) agaings the application, and the set of full nodes 
+(we call them peers) that are block providers.
+
+Peers can be faulty, and we don't make any assumption about rate of correct/faulty nodes in the
+node peerset (i.e., they can all be faulty). Processes (client and peers) communicate
+by message passing. They exchange the following messages:
+
+- StatusRequest
+- StatusReply
+- BlockRequest
+- BlockReply
+ 
+Correct peers are part of the replicated state machine, i.e., they manage
+blockchain and execute transactions against the same deterministic application A.
+We assume Tendermint failure model, i.e., given a trusted header h, we assume that >2/3 of voting power will behave
+correctly until time h.Time + TRUSTED_PERIOD. We don't make any assumptions about behavior of faulty processes.
+
+In this spec we assume a simpler problem (compared to complete fast-sync) defined with the following assumptions: 
+
+- node is initialised with a set of peers with which messages can be exchanged
+- peers can be removed (in case they misbehave) but they cannot be added
+- some numbers of peers is faulty (correct and faulty peer ids are input parameter)
+- peers do not modify its height, i.e., we take a snaphsot of the system and try to sync as much as we can
+within that shapshot; therefore StatusRequest message is sent only once
+- we assume synchronous communication between correct processes (node and correct peers) and model communication
+with asycnhronous rpc calls, i.e, every request receives corresponding reply 
+- faulty processes can send arbitrary messages (even if they are not requested)
+- fast sync should terminate with at least a block height that is equal to the maximum height of the correct peer
+
+This spec can be extended allong the following lines to capture full fast sync problem:
+
+1) Node can establish connections to other peers during fast sync protocol. This should probably be modelled by having
+additional input message received by Node (AddPeer). This should be relatively easily supported
+as it shouldn't significantly affect termination conditions. More preciselly, the properties should be very similar 
+in case we bound eventually reception of AddPeer messages; otherwise we can't terminate as AddPeer message can 
+always delay termination.
```

Comment:
In case you are fast syncing from scratch, it can be quite long (@ancazamfir  and @brapse did experiments). However, after state sync is in place, fast sync should probably execute only after state sync is over so it should terminate faster.